### DISCUSSION TITLE
Allow active property in gradle to take booleans

### DISCUSSION THREAD

**Discussion by crschnick** [jreleaser/jreleaser#1320] [Ideas]
If you have your own logic for enabling or disabling something, I would like to assign the active property with a boolean. Currently I just do `active = <expr> ? 'ALWAYS' : 'NEVER'`.

**Comment by aalmiray** [comment]
As it happens, we used to have `enabled` as a boolean property in many DSL elements. We decided to switch to `active` to be able to have more than 2 possible states as the property could be set based on the project's "releasable" state as well: `RELEASE`, `PRERELEASE`.

**Comment by crschnick** [comment]
Is it possible to create your own states?

**Reply by aalmiray** [reply]
No. It's not possible as the current values are defined by an enum. Yet, you can use build script logic to assign your custom state to any of the 4 possible values.

**Comment by crschnick** [comment]
Right now I am dealing with creating staging releases and they don't really fit any of the four values.

**Reply by aalmiray** [reply]
The four values mean:
- ALWAYS. active no matter what. 
- NEVER. inactive no matter what. 
- PRERELEASE. active if the project version matches its snapshot pattern. 
- RELEASE. active if the project version is not snapshot.

**Reply by crschnick** [reply]
In my case a staging release is not a snapshot but is still handled differently (e.g. no announcement, different release repo, etc.) compared to the proper release, so I currently just use my own `isStagingRelease` gradle property. I mean it works, but doesn't look as pretty.

**Reply by aalmiray** [reply]
Do you need staged and regular releases? Sharing common configuration?

**Reply by crschnick** [reply]
Yes, I use staged releases to a separate repository such that potential users don't accidentally download them. They are mainly used for testing things in production.
